mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -197,3 +197,28 @@ class ChiSqSelector @Since("1.3.0") (
     new ChiSqSelectorModel(indices)
   }
 }
+
+/**
+ * Creates a ChiSquared feature selector by False Positive Rate (FPR) test.
+ * @param alpha the highest p-value for features to be kept
+ */
+@Since("2.1.0")
+class ChiSqSelectorByFpr @Since("2.1.0") (
+  @Since("2.1.0") val alpha: Double) extends Serializable {
+
+  /**
+   * Returns a ChiSquared feature selector by FPR.
+   *
+   * @param data an `RDD[LabeledPoint]` containing the labeled dataset with categorical features.
+   *             Real-valued features will be treated as categorical for each distinct value.
+   *             Apply feature discretizer before using this function.
+   */
+  @Since("2.1.0")
+  def fit(data: RDD[LabeledPoint]): ChiSqSelectorModel = {
+    val indices = Statistics.chiSqTest(data)
+      .zipWithIndex.filter { case (res, _) => res.pValue < alpha }
```

Comment:
@srowen I've checked our thread with @mengxr on that feature https://github.com/apache/spark/pull/1484. 
  - We preserve the order of indexes to make the selection of features with one loop (i.e. linear time complexity). Here is the code: https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala#L74. The logic of feature selector, which is selection of N top features, does not imply that it will sort the features by their Chi-square value. A parameter must be introduced if it is required for some use-case.
  - We were planning to include Chi-square values in the model later if needed https://github.com/apache/spark/pull/1484#discussion_r23876952

@mpjlu It seems that FPR feature selection should not modify the code of existing `ChiSqSelector`, because FPR feature selection works on top of a scoring function rather than on top of another selector. Scoring function is a parameter, and it might be Chi-square. For example, please refer to Sklearn's `FPR` implementation mentioned. It uses ANOVA as a default scoring function http://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.SelectFpr.html#sklearn.feature_selection.SelectFpr.